pallets/pallet-bonded-coins/src/curves/lmsr.rs

```diff
@@ -49,48 +63,24 @@ where
 	Parameter: FixedSigned + PartialOrd<Precision> + From<Precision> + ToFixed,
 	<Parameter as Fixed>::Bits: Copy + ToFixed + AddAssign + BitOrAssign + ShlAssign,
 {
-	// c(a, c) = (a - c) + b * ln((1 + SUM_i e^((q_i - a)/b)) / (1 + SUM_i e^((q_i - c)/b)))
 	fn calculate_costs(
 		&self,
 		low: Parameter,
 		high: Parameter,
-		passive_supply: PassiveSupply<Parameter>,
+		mut passive_supply: PassiveSupply<Parameter>,
```

Comment:
I don't think it's wise to modify an input parameter in a function that calculate costs. I would suggest the input `passive_supply` to not be `mut`.

Reply:
Is this an issue? the function takes over the overship of the variable

Reply:
[75be462](https://github.com/KILTprotocol/kilt-node/pull/777/commits/75be462022dada2193acf79c528d6118d46e2020)

Reply:
In a function that calculates something, I think it's misleading to mutate the input parameter, yes.